cmd/login/wc.go

```
@@ -147,65 +200,90 @@ func (r *runner) createClusterClientCert(ctx context.Context, client k8sclient.I
+func (r *runner) storeWCCredentials(c credentialConfig) (string, bool, error) {
 
-	if r.loginOptions.selfContainedWC {
-		fmt.Fprintf(r.stdout, "A new kubectl context has been created named '%s' and stored in '%s'. You can select this context like this:\n\n", contextName, r.flag.SelfContained)
-		fmt.Fprintf(r.stdout, "  kubectl cluster-info --kubeconfig %s \n", r.flag.SelfContained)
-	} else if !r.loginOptions.switchToWCcontext {
-		fmt.Fprintf(r.stdout, "A new kubectl context has been created named '%s'. To switch back to this context later, use this command:\n\n", contextName)
-		fmt.Fprintf(r.stdout, "  kubectl config use-context %s\n", contextName)
-	} else if contextExists {
-		fmt.Fprintf(r.stdout, "Switched to context '%s'.\n\n", contextName)
-		fmt.Fprintf(r.stdout, "To switch back to this context later, use this command:\n\n")
-		fmt.Fprintf(r.stdout, "  kubectl config use-context %s\n", contextName)
-	} else {
-		fmt.Fprintf(r.stdout, "A new kubectl context has been created named '%s' and selected. To switch back to this context later, use this command:\n\n", contextName)
-		fmt.Fprintf(r.stdout, "  kubectl config use-context %s\n", contextName)
+	// Store client certificate credential either into the current kubeconfig or a self-contained file if a path is given.
+	if r.loginOptions.selfContainedWC && c.filePath != "" {
+		return printWCCredentials(r.k8sConfigAccess, r.fs, c)
 	}
-
-	return nil
+	return storeWCCredentials(r.k8sConfigAccess, r.fs, c)
 }
 
-func getClusterBasePath(k8sConfigAccess clientcmd.ConfigAccess) (string, error) {
+func getClusterBasePath(k8sConfigAccess clientcmd.ConfigAccess, provider string) (string, error) {
 	config, err := k8sConfigAccess.GetStartingConfig()
 	if err != nil {
 		return "", microerror.Mask(err)
 	}
 
 	clusterServer, _ := kubeconfig.GetClusterServer(config, config.CurrentContext)
+	clusterName := kubeconfig.GetCodeNameFromKubeContext(config.CurrentContext)
```

This function should maybe return an error if the context doesn't match the expected format. Also the kubeconfig from LastPass has a context that looks like `gamma-admin@gamma` which wouldn't work here (it seems to only accept the SSO context like `gs-gamma`).

---

Good point! :)
I renamed the gamma context and didn't catch that one.